### DISCUSSION TITLE
AM243x MCU+ SDK

### DISCUSSION THREAD

**Discussion by kkitayam** [hathach/tinyusb#1098] [General]
[AM243x MCU+ SDK document](https://software-dl.ti.com/mcu-plus-sdk/esd/AM243X/latest/exports/docs/api_guide_am243x/index.html) describes that TinyUSB is included as a USB stack. This is very interesting.
Has anyone tried this?

**Comment by hathach** [comment]
Interesting, this is new to me as well. AM243x is a beast supporting USB 3.1 with 5Gbps PHY which is not seen so far on any MCUs. I just check out its reference manual. Unfortunately, the register interface details is not publicly available. It is probably Synopsys DWC3, I will try to download its sdk and take a look around. 

```
Details on the USB controller registers are not provided in this TRM. The existing Linux or RTOS
drivers should be used for proper USB operation. For those who do wish to substantially modify the
existing Linux or RTOS USB driver(s), or create new drivers, contact TI for more information on how to
obtain the third-party documentation under NDA.
```

**Reply by kkitayam** [reply]
Yeah, it is bad hardware details are not public.

[USB sction](https://software-dl.ti.com/mcu-plus-sdk/esd/AM243X/latest/exports/docs/api_guide_am243x/USB_DEVICE_DRIVER.html#autotoc_md648) of SDK document mentions using Cadence USB core driver. It may be cadence [USB IP](https://ip.cadence.com/ipportfolio/ip-portfolio-overview/interface-ip/usb-ip/usb-3-0-controller-2/usb-3-0-drd-controller).

**Reply by hathach** [reply]
great finding, I totally miss that, in that case, this is probably the driver for it in the linux tree https://github.com/torvalds/linux/tree/master/drivers/usb/cdns3 . It probably takes awhile for me get the board and find time testing this out. But it is great to see tinyusb is integrated to more vendor's sdk. Especially TI has their own usb stack for many of their mcues :).

**Comment by hathach** [comment]
AM243x LaunchPad is available here https://www.ti.com/store/ti/en/p/product/?p=LP-AM243 for 89$ . It is good price devkit of 800mhz mcu with gigabit ethernet + USB and on-chip 2.0 phy + pcie. Though since the USB controller requires NDA, I am not sure if we could actually do anything with it.
